### ISSUE TITLE
Time.zone.parse cannot always replace DateTime.parse

### ISSUE THREAD

**Issue by monfresh** [bbatsov/rubocop#1778]
In the following cases, the Rubocop suggestion would break functionality:
```ruby
DateTime.parse('Monday').strftime('%u').to_i
 => 1

Time.zone.parse('Monday').strftime('%u').to_i
 => 4

DateTime.parse('Monday').wday
 => 1

Time.zone.parse('Monday').wday
 => 4
```

**Comment by palkan** [comment]
Unfortunately, Rails `Time.zone.parse` cannot handle weekday-only strings correctly (it always returns `today` instead).

And I don't think we should handle such a rare case in Rails/TimeZone too. Just disable the cop for this line.

**Comment by aripollak** [comment]
I noticed the DateTime cop also triggering incorrectly for `DateTime.iso8601`. ISO8601 format includes timezone information, so `Time.zone.iso8601` shouldn't make a difference even that did exist.
